src/Microsoft.AspNet.Security/AuthenticationTicket.cs

```diff
@@ -29,11 +29,33 @@ public AuthenticationTicket(ClaimsIdentity identity, AuthenticationProperties pr
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationTicket"/> class
+        /// </summary>
+        /// <param name="identity"></param>
+        /// <param name="properties"></param>
+        public AuthenticationTicket(ClaimsPrincipal principal, AuthenticationProperties properties, string authenticationType)
+        {
+            AuthenticationType = authenticationType;
+            Principal = principal;
+            Properties = properties ?? new AuthenticationProperties();
+        }
+
+        /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public string AuthenticationType { get; private set; }
+
+        /// <summary>
         /// Gets the authenticated user identity.
         /// </summary>
         public ClaimsIdentity Identity { get; private set; }
```

Comment:
I agree with @Tratcher, having a `Principal` here (whose associated comment is wrong BTW) seems really weird.